metricbeat/module/elasticsearch/elasticsearch.go

```diff
@@ -285,3 +341,26 @@ func (c *_licenseCache) set(license common.MapStr, ttl time.Duration) {
 	c.ttl = ttl
 	c.cachedOn = time.Now()
 }
+
+func getSettingGroup(allSettings common.MapStr, groupKey string) (common.MapStr, error) {
+	hasSettingGroup, err := allSettings.HasKey(groupKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "failure to determine if "+groupKey+" settings exist")
+	}
+
+	if !hasSettingGroup {
+		return common.MapStr{}, nil
```

Comment:
Do you want to return here nil or an empty MapStr?

Reply:
If we return `nil` here, then on the caller side we'll have to check for it and construct an empty MapStr. Otherwise this line will throw a runtime panic:

https://github.com/elastic/beats/blob/d38f1dd5801ab4fb7fb320aedc504f5e4ea060c2/metricbeat/module/elasticsearch/elasticsearch.go#L302

Reply:
Thinking about the end JSON document, what is your suggestion on what should be there?

Reply:
I decided that if there are no cluster settings, we should not have that field in the end JSON document either. This also makes it consistent with the internal ES collector implementation:

https://github.com/elastic/elasticsearch/blob/b0131399cbd85f6cbd1ac9cff4264929da54629c/x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/collector/cluster/ClusterStatsMonitoringDoc.java#L169-L184

I've implemented this logic in 43fed80c8.